agenta-backend/agenta_backend/routers/observability_router.py

```diff
@@ -1,99 +1,163 @@
-import os
-from typing import List
+from typing import List, Any
 
-from fastapi import Request
-from agenta_backend.utils.common import APIRouter
+from fastapi import Request, Query, Depends
 
-from agenta_backend.services.event_db_manager import (
-    get_variant_traces,
-    create_app_trace,
-    create_trace_span,
-    get_trace_single,
-    trace_status_update,
-    get_trace_spans,
-    add_feedback_to_trace,
-    get_trace_feedbacks,
-    get_feedback_detail,
-    update_trace_feedback,
+from agenta_backend.utils.common import APIRouter
+from agenta_backend.services import event_db_manager, redis_cache_service
+from agenta_backend.models.api.api_models import (
+    WithPagination,
+    SorterParams,
+    PaginationParam,
 )
 from agenta_backend.models.api.observability_models import (
-    Span,
+    SpanDetail,
     CreateSpan,
     CreateFeedback,
     Feedback,
     UpdateFeedback,
-    Trace,
+    TraceDetail,
     CreateTrace,
     UpdateTrace,
+    GenerationFilterParams,
+    ObservabilityDashboardDataRequestParams,
 )
 
 
 router = APIRouter()
 
 
+@router.get(
+    "/dashboard/",
+    operation_id="observability_dashboard",
+)
+async def get_dashboard_data(
+    request: Request,
+    app_id: str,
+    parameters: ObservabilityDashboardDataRequestParams = Depends(),
+):
+    try:
+        dashboard_data = await redis_cache_service.cache_observability_data(
+            event_db_manager.retrieve_observability_dashboard,
+            **{
+                "app_id": app_id,
+                "parameters": parameters,
+                "cache_key": str(request.url),
+            }
+        )
+        return dashboard_data
+    except Exception as e:
+        import traceback
+
+        traceback.print_exc()
+        return []
+
+
 @router.post("/traces/", response_model=str, operation_id="create_trace")
```

Comment:
See my comment in slack. I think we should move to the otel implementation, post traces takes a list of spans.